c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -688,6 +753,61 @@ public:
     });
   }
 
+  kj::Promise<void> connect(ConnectContext context) override {
+    auto params = context.getParams();
+    auto host = params.getHost();
+    auto headers = factory.headersToKj(params.getHeaders());
+    auto upPipe = kj::newOneWayPipe();
+
+    PipelineBuilder<ConnectResults> pb;
+    pb.setUp(factory.streamFactory.kjToCapnp(kj::mv(upPipe.out)));
+
+    context.setPipeline(pb.build());
+
+    auto client = kj::newHttpClient(*inner);
```

Comment:
Hmm I hate to say this but we really shouldn't be using a client adapter here. The http-over-capnp interface is really most analogous to `HttpService` rather than `HttpClient`, so trying to implement it in terms of `HttpClient` is probably making the code below significantly more complex than it would otherwise be. Additionally, using a client adapter adds a fair amount of overhead inside the adapter, as well as at least one known bug (the fact that it cancels the service prematurely at the end of the stream).

If you look at `request()`/`startRequest()`, they are implemented directly in terms of `HttpService`, we should use the same model here.

Sorry, I really should have noticed this when this was a draft... I remember you had a few problems that involved the client adapter but for some reason I didn't connect the dots.